command_test.go

```diff
@@ -89,3 +108,49 @@ Commands:
 		assert.Equal(t, t.Name(), expected, got)
 	})
 }
+
+func TestSubCmdHelpOutput(t *testing.T) {
+	withNested := `Usage: mockSubCmd Test a subcommand.
+
+Aliases: s, sc, sub
+
+Description: A simple mock subcommand with aliases and its own subcommand.
+
+Commands:
+	s, sc, sub, mockSubCmd  - A simple mock subcommand with aliases and its own subcommand.
+`
+
+	noNested := `Usage: mockSubCmdNoNested Used for help output tests.
+
+Aliases: s, sc, sub
+
+Description: A simple mock subcommand with aliases and no nested subcommands.
+`
+
+	for _, test := range []struct {
+		cmd            Command
+		name, expected string
+	}{
+		{
+			name:     "subcmd w/nested subcmd.",
+			expected: withNested,
+			cmd:      new(mockSubCmd),
+		},
+		{
+			name:     "subcmd w/no nested subcmds.",
+			expected: noNested,
+			cmd:      new(mockSubCmdNoNested),
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			buf := new(bytes.Buffer)
+			name := test.cmd.Spec().Name
+			fl := pflag.NewFlagSet(name, pflag.ExitOnError)
+			// If the help output is not written to the buffer
+			// in the format we expect then the test will fail.
+			renderHelp(name, test.cmd, fl, buf)
```

Comment:
outside the scope of this PR, but by convention, the writer is usually the first param (or second if there is a context) like `fprintf` & co.